raiden/raiden_service.py

```diff
@@ -204,7 +211,7 @@ def __init__(self, chain, private_key_bin, transport, discovery, config):
         self.swapkey_to_greenlettask = dict()
 
         self.identifier_to_statemanagers = defaultdict(list)
-        self.identifier_to_results = defaultdict(list)
+        self.identifier_to_results = dict()
```

Comment:
rename the variable from `identifier_to_results` to `identifier_to_result`, we use plural to say the mapping is to a list of values.

PS: I'm not sure if this is a good change.

Reply:
Ok, I'll make the renaming. The point is, we had already stored the `async_result` for mediated transfers. I'm just expanding the behavior to direct transfers too, and making it unique (unique `identifier`).